### PR TITLE
Easily generate all-quarkus-properties.json

### LINKED ISSUE #182: Easily generate all-quarkus-properties.json
Currently, there is an [all-quarkus-properties.json](https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.ls/com.redhat.microprofile.ls/src/test/resources/com/redhat/microprofile/services/all-quarkus-properties.json) file that contains a JSON array of location definiton in the format of [`PropertyDefinition[]`](https://github.com/redhat-developer/quarkus-ls/blob/c954da4987611102a2a98e69549425e0bbe1e258/microprofile.ls/com.redhat.microprofile.ls/src/test/java/com/redhat/microprofile/ls/MockMicroProfilePropertyDefinitionProvider.java#L32-L46)

As mentioned in this comment: https://github.com/redhat-developer/quarkus-ls/pull/177#issuecomment-571221935, it would be great if there was a new class that is responsible for generating these definitions and change the definition format.

### PR DESCRIPTION
**Pull request by angelozerr**
Easily generate all-quarkus-properties.json

Fixes #182

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: microprofile.jdt/com.redhat.microprofile.jdt.test/src/main/java/com/redhat/microprofile/jdt/core/GenerateAllPropertiesAndDefinition.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
+package com.redhat.microprofile.jdt.core;
```
Comment (fbricon) [inline microprofile.jdt/com.redhat.microprofile.jdt.test/src/main/java/com/redhat/microprofile/jdt/core/GenerateAllPropertiesAndDefinition.java]: header

File: microprofile.jdt/com.redhat.microprofile.jdt.test/src/main/java/com/redhat/microprofile/jdt/core/GenerateAllPropertiesAndDefinition.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
+/*******************************************************************************
+* Copyright (c) 2019 Red Hat Inc. and others.
```
Comment (fbricon) [inline microprofile.jdt/com.redhat.microprofile.jdt.test/src/main/java/com/redhat/microprofile/jdt/core/GenerateAllPropertiesAndDefinition.java]: 2020

**Comment by fbricon** [comment]
Build fails

**Comment by angelozerr** [comment]
Please not ethis PR depends on https://github.com/redhat-developer/quarkus-ls/pull/197.

Once https://github.com/redhat-developer/quarkus-ls/pull/197 will be merged, I will rebase this PR to the master

**Comment by angelozerr** [comment]
@fbricon please review the PR. Now the tests of MicroProfile LS uses the generated JSON files for test

File: microprofile.jdt/com.redhat.microprofile.jdt.test/src/main/java/com/redhat/microprofile/jdt/core/GenerateAllPropertiesAndDefinition.java [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+import com.redhat.microprofile.commons.metadata.ItemHint.ValueHint;
+import com.redhat.microprofile.commons.metadata.ItemMetadata;
+import com.redhat.microprofile.jdt.core.utils.IJDTUtils;
+import com.redhat.microprofile.jdt.internal.core.ls.JDTUtilsLSImpl;
+
+/**
+ * Generator to generate JSON properties and definitions used for Junit test of
+ * MicroProfile LS.
+ * 
+ * @author Angelo ZERRo
```
Comment (fbricon) [inline microprofile.jdt/com.redhat.microprofile.jdt.test/src/main/java/com/redhat/microprofile/jdt/core/GenerateAllPropertiesAndDefinition.java]: Zerr or Zero?

Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.test/src/main/java/com/redhat/microprofile/jdt/core/GenerateAllPropertiesAndDefinition.java]: @fbricon fixed :)

**Pull request merged**